src/libs/ReportUtils.ts

```diff
@@ -216,294 +488,218 @@ function isCanceledTaskReport(report = {}, parentReportAction = {}) {
+    return isTaskReport(report) && !isCanceledTaskReport(report, parentReportAction) && report?.stateNum === CONST.REPORT.STATE_NUM.OPEN && report?.statusNum === CONST.REPORT.STATUS.OPEN;
 }
 
 /**
  * Checks if a report is a completed task report.
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isCompletedTaskReport(report) {
-    return isTaskReport(report) && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isCompletedTaskReport(report: OnyxEntry<Report>): boolean {
+    return isTaskReport(report) && report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report?.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Checks if the current user is the manager of the supplied report
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportManager(report) {
-    return report && report.managerID === currentUserAccountID;
+function isReportManager(report: OnyxEntry<Report>): boolean {
+    return report?.managerID === currentUserAccountID;
 }
 
 /**
  * Checks if the supplied report has been approved
- *
- * @param {Object} report
- * @returns {Boolean}
  */
-function isReportApproved(report) {
-    return report && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
+function isReportApproved(report: OnyxEntry<Report>): boolean {
+    return report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && report.statusNum === CONST.REPORT.STATUS.APPROVED;
 }
 
 /**
  * Given a collection of reports returns them sorted by last read
- *
- * @param {Object} reports
- * @returns {Array}
  */
-function sortReportsByLastRead(reports) {
-    return _.chain(reports)
-        .toArray()
-        .filter((report) => report && report.reportID && report.lastReadTime)
-        .sortBy('lastReadTime')
-        .value();
+function sortReportsByLastRead(reports: OnyxCollection<Report>): Array<OnyxEntry<Report>> {
+    return Object.values(reports ?? {})
+        .filter((report) => report?.reportID && report?.lastReadTime)
+        .sort((a, b) => {
+            const aTime = new Date(a?.lastReadTime ?? '');
+            const bTime = new Date(b?.lastReadTime ?? '');
+
+            return aTime.valueOf() - bTime.valueOf();
+        });
 }
 
 /**
  * Whether the Money Request report is settled
- *
- * @param {String} reportID
- * @returns {Boolean}
  */
-function isSettled(reportID) {
+function isSettled(reportID: string | undefined): boolean {
     if (!allReports) {
         return false;
     }
-    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] || {};
-    if ((typeof report === 'object' && Object.keys(report).length === 0) || report.isWaitingOnBankAccount) {
+    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+    if ((typeof report === 'object' && isEmptyObject(report)) || report?.isWaitingOnBankAccount) {
```

Comment:
```suggestion
    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`] ?? {};
    if (isEmptyObject(report) || report?.isWaitingOnBankAccount) {
```
Maybe this is more accurate

Reply:
why do you want to default to {} ? without that type of reprot is Report | null so actually we can do it as 
```suggestion
    const report = allReports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
    if (!!report?.isWaitingOnBankAccount) {
```

Reply:
I think `isEmptyObject` should funnel down Nullable types too